### PR TITLE
1219 patient item and address attributes 

### LINKED ISSUE #1219: Add and extract attributes to Patient model
Currently, in the Quest app, there is a Patient model that only has the following attributes: id, identifier, name, gender, age, address, & additional data. However, for the Tingathe app, we need some additional attributes that we are saving to the server.

When looking at the Patient FHIR specification, we need the following additional attributes:

- telcom
- address
- contact
- generalPractitioner
- managingOrganization
- link

The Patient model does have an address attribute but it is extracted as a string as a result of a combination of the individual address attributes. However, we need to be able to access the individual attributes within the address itself. Specifically, we need to access:

- district
- state
- text

### PR DESCRIPTION
**Pull request by AbdulWahabMemon**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #1219

**Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/PatientExtension.kt [hunk 1]
```diff
@@ -135,6 +135,36 @@ fun Patient.extractAddress(): String {
...
+
+fun Patient.extractAddressText(): String {
+  if (!hasAddress()) return ""
+  return with(addressFirstRep) { this.text ?: "" }
+}
+
+fun Patient.extractTelecom(): List<String>? {
+  if (!hasTelecom()) return null
+  return telecom.map { it.value }
+}
```
Comment (ellynerdstone) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/PatientExtension.kt:156]: It's possible to have these implementation in one function and use utility class `org.smartregister.fhircore.engine.util.fhirpath.FhirPathDataExtractor` to extract the relevant properties from the patient resource.

Comment (ellykits) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/PatientExtension.kt:156]: It's possible to have these implementation in one function and use utility class `org.smartregister.fhircore.engine.util.fhirpath.FhirPathDataExtractor` to extract the relevant properties from the patient resource.

**Comment by ellykits** [comment]
@AbdulWahabMemon I think we can close this. As you are working on the mwcore register implementation in another branch; this functionality can be added to your other branch.

**Comment by AbdulWahabMemon** [comment]
> @AbdulWahabMemon I think we can close this. As you are working on the mwcore register implementation in another branch; this functionality can be added to your other branch.

The requested changes with FhirPathDataExtractor also done in another branch, so we can close this one.

**Pull request closed without merging**